### ISSUE TITLE
QR Image not generated on iOS < 13

### ISSUE THREAD

**Issue by Jorundur** [gevgasparyan/rn-qr-generator#3]
Thanks for a good package, it's works well.

However, I'm having some issues when using iOS 12 (also tried iOS 11) where the generated QR image is `nil`.

This is not an issue with iOS 13 - the library works as it should both in simulator and when building to a device when on iOS 13.

The issue seems to be this line (or at least it manifests itself there) https://github.com/gevgasparyan/rn-qr-generator/blob/master/ios/RNQrGenerator.m#L57

i.e. `NSData *qrData = UIImagePNGRepresentation(image);` - I've been debugging this and even though `image` is defined here, `UIImagePNGRepresentation(image)` returns `nil` so `qrData` is `nil` for the rest of the program.

This only happens in iOS 12 (both when using simulator and building to device) but not in iOS 13.

Do you have any ideas how this can be fixed?

**Comment by Jorundur** [comment]
Quick update,

I found a fix for this that makes it work in both iOS 12 and iOS 13 (haven't tried older than 12). When I have time I might do a PR if you think this is a good idea.

Basically I changed line 55 in RNQrGenerator from:

```objc
UIImage *image = [UIImage imageWithCIImage:qrImage];
```

to:
```objc
CIContext *context = [CIContext contextWithOptions:nil];
CGImageRef cgImage = [context createCGImage:qrImage fromRect:[qrImage extent]];
UIImage *image = [UIImage imageWithCGImage:cgImage];
```

(Idea from this post: https://bitwordblog.wordpress.com/2018/01/25/generate-qrcode-and-barcode-in-objective-c/)

**Comment by gevgasparyan** [comment]
Thanks, @Jorundur .

Feel free to open a PullRequest. It's a good idea

**Comment by gevgasparyan** [comment]
fixed in version 1.0.3
